pkg/console/status/auth_status.go

```diff
@@ -0,0 +1,202 @@
+	now := metav1.Now()
+	c.setCondition(ConditionTypeAvailable, metav1.ConditionTrue, reason, message, now)
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+func (c *AuthStatusHandler) setCondition(conditionType AuthenticationConditionType, status metav1.ConditionStatus, reason, message string, ts metav1.Time) *AuthStatusHandler {
+	if c.conditionsToApply[conditionType] == nil {
+		c.conditionsToApply[conditionType] = &metav1.Condition{Type: string(conditionType)}
+	}
+
+	c.conditionsToApply[conditionType].Status = status
+	c.conditionsToApply[conditionType].Reason = reason
+	c.conditionsToApply[conditionType].Message = message
+	c.conditionsToApply[conditionType].LastTransitionTime = ts
+
+	return c
+}
+
+func (c *AuthStatusHandler) WithCurrentOIDCClient(currentClientID string) *AuthStatusHandler {
+	c.currentClientID = currentClientID
+	return c
+}
+
+func (c *AuthStatusHandler) Apply(ctx context.Context, authnConfig *configv1.Authentication) error {
+	if authnConfig.Spec.Type != configv1.AuthenticationTypeOIDC {
+		return fmt.Errorf("authentication type not OIDC")
+	}
+
+	applyConfig, err := configv1ac.ExtractAuthenticationStatus(authnConfig, c.fieldManager)
+	if err != nil {
+		return err
+	}
+
+	if applyConfig.Status == nil {
+		applyConfig.Status = &configv1ac.AuthenticationStatusApplyConfiguration{}
+	}
+
+	idx := -1
+	for i, client := range applyConfig.Status.OIDCClients {
+		if *client.ComponentName == c.componentName && *client.ComponentNamespace == c.componentNamespace {
+			idx = i
+			break
+		}
+	}
+
+	// create the OIDCClientStatus for the operator if it doesn't exist
+	if idx == -1 {
+		if applyConfig.Status.OIDCClients == nil {
+			applyConfig.Status.OIDCClients = make([]configv1ac.OIDCClientStatusApplyConfiguration, 0)
+		}
+
+		applyConfig.Status.OIDCClients = append(applyConfig.Status.OIDCClients, configv1ac.OIDCClientStatusApplyConfiguration{
+			ComponentName:      &c.componentName,
+			ComponentNamespace: &c.componentNamespace,
+		})
+		idx = len(applyConfig.Status.OIDCClients) - 1
+	}
+
+	// if a client is available, create the relevant entry in the CurrentOIDCClients if none exists
+	if len(c.currentClientID) > 0 {
+		clientFound := false
+		for _, client := range authnConfig.Spec.OIDCProviders[0].OIDCClients {
+			if client.ComponentName == c.componentName && client.ComponentNamespace == c.componentNamespace && client.ClientID == c.currentClientID {
+				clientFound = true
+				break
+			}
+		}
+
+		if clientFound {
+			// a client has been configured in the spec; use it to update CurrentOIDCClients
+			oidcClientRef := configv1ac.OIDCClientReferenceApplyConfiguration{
+				OIDCProviderName: &authnConfig.Spec.OIDCProviders[0].Name,
+				IssuerURL:        &authnConfig.Spec.OIDCProviders[0].Issuer.URL,
+				ClientID:         &c.currentClientID,
+			}
+
+			currentClientFound := false
```

Comment:
always set it

Reply:
Will there always be at most one `CurrentOIDCClient` in the list?